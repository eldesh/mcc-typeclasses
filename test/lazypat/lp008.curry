-- Here is a canonical example for using right hand side constraints in a
-- data type declaration. These constraints are used to dynamically
-- provide the corresponding type class instances in the scope of the
-- data constructor. Note that this does not extend to lazy patterns.
-- (Also see tests lp005.curry, lp006.curry and lp007.curry for variants
-- that use existential quantification.)

data Showable a = Show a => Showable a

f1 (Showable x) = Just (show x)
f2 ~(Showable x) = Just (show x)
f3 s = Just (show x) where Showable x = s



