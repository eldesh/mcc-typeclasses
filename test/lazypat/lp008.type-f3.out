Show _1 => Showable _1 -> Maybe [Char]
